xtreeshr/XTreeGetTimedRecord.c

```diff
@@ -341,7 +341,11 @@ EXPORT int XTreeGetTimedRecord(int inNid, mdsdsc_t *startD,
 
     MdsFree1Dx(&dataXds[currSegIdx], 0);
     MdsFree1Dx(&dimensionXds[currSegIdx], 0);
-    signals[currSegIdx] = (mds_signal_t *)resampledXds[currSegIdx].pointer;
+//After resampling some segments may be empty: consider only nonempty segments
+    if(resampledXds[currSegIdx].l_length > 0)
+      signals[nonEmptySegIdx++] = (mds_signal_t *)resampledXds[currSegIdx].pointer;
+    else //Keep track that the segment is not considered
+      signalsApd.arsize -= sizeof(mds_signal_t *);
```

Comment:
shoukd it not better be using the length field of signalsApd instead of sizeof. best would be though to calculate arsize based on nonEmptySegIdx an length after the for loop.

Reply:
we should add some tests that tackle those kind of edge cases because xtreeshr is used frequently and its a pitty its not tested well. i may be able to contribute. we should add these tests to treeUnitTest where i test the timecontext stuff. maybe there is a lightweight example that would trigger the issue w/o preexisting tree files.

Reply:
Ok, removed length update after the loop